hoomd/Trigger.h

```diff
@@ -87,5 +103,176 @@ class PYBIND11_EXPORT PeriodicTrigger : public Trigger
         uint64_t m_phase;
     };
 
+/** Until trigger
+ *
+ *  Trigger every time step until `timestep >= m_until`.
+*/
+class PYBIND11_EXPORT UntilTrigger : public Trigger
+    {
+    public:
+
+    UntilTrigger(uint64_t until) : Trigger(), m_until(until) {}
+
+    bool compute(uint64_t timestep)
+        {
+        if (timestep < m_until)
+            {
+            return true;
+            }
+        else
+            {
+            return false;
+            }
+        }
+
+    /// Get the point where triggering will stop
+    uint64_t getUntil() {return m_until;}
+
+    /// Set the point where triggering will stop
+    void setUntil(uint64_t until) {m_until = until;}
+
+    protected:
+        /// Always trigger until this timestep
+        uint64_t m_until;
+    };
+
+/** After trigger
+ *
+ *  Trigger every time step after and including `timestep > m_after`.
+*/
+class PYBIND11_EXPORT AfterTrigger : public Trigger
+    {
+    public:
+
+    AfterTrigger(uint64_t after) : Trigger(), m_after(after) {}
+
+    bool compute(uint64_t timestep)
+        {
+        if (timestep > m_after)
+            {
+            return true;
+            }
+        else
+            {
+            return false;
+            }
+        }
+
+    /// Get the point where triggering will stop
+    uint64_t getAfter() {return m_after;}
+
+    /// Set the point where triggering will stop
+    void setAfter(uint64_t after) {m_after = after;}
+
+    protected:
+        /// Always trigger after this timestep
```

Comment:
Make this comment specify "including" or not, depending on how you resolve other comments.